src/search/pdbs/pattern_collection_generator_hillclimbing.cc

```diff
@@ -389,30 +411,39 @@ void PatternCollectionGeneratorHillclimbing::hill_climbing(
             // Remove the added PDB from candidate_pdbs.
             candidate_pdbs[best_pdb_index] = nullptr;
 
-            utils::g_log << "Hill climbing time so far: "
-                         << hill_climbing_timer->get_elapsed_time()
-                         << endl;
+            if (verbosity >= utils::Verbosity::VERBOSE) {
+                utils::g_log << "Hill climbing time so far: "
+                             << hill_climbing_timer->get_elapsed_time()
+                             << endl;
+            }
         }
     } catch (HillClimbingTimeout &) {
-        utils::g_log << "Time limit reached. Abort hill climbing." << endl;
+        if (verbosity >= utils::Verbosity::NORMAL) {
+            utils::g_log << "Time limit reached. Abort hill climbing." << endl;
+        }
     }
 
-    utils::g_log << "Hill climbing iterations: " << num_iterations << endl;
-    utils::g_log << "Hill climbing generated patterns: " << generated_patterns.size() << endl;
-    utils::g_log << "Hill climbing rejected patterns: " << num_rejected << endl;
-    utils::g_log << "Hill climbing maximum PDB size: " << max_pdb_size << endl;
-    utils::g_log << "Hill climbing time: "
-                 << hill_climbing_timer->get_elapsed_time() << endl;
+    if (verbosity >= utils::Verbosity::NORMAL) {
+        utils::g_log << "Hill climbing iterations: " << num_iterations << endl;
+        utils::g_log << "Hill climbing generated patterns: " << generated_patterns.size() << endl;
+        utils::g_log << "Hill climbing rejected patterns: " << num_rejected << endl;
+        utils::g_log << "Hill climbing maximum PDB size: " << max_pdb_size << endl;
+        utils::g_log << "Hill climbing time: "
+                     << hill_climbing_timer->get_elapsed_time() << endl;
```

Comment:
I prefer chaining << over repeating utils::g_log

Reply:
I prefer repeating the stream after each endl, i.e., I dislike chaining over `endl`. I couldn't find anything regarding this in our coding conventions, but I think I have never seen something like `cout << "bla" << endl << "foo" << endl` in Fast Downward.